### PR TITLE
feat(core): add easy importValue to CfnOutput

### LINKED ISSUE #11360: [cdk-pipelines] Cannot add codepipeline action within an Application Stage
### Adding an action within an Application Stage
I am trying to add an application stage to my pipeline with everything required for deploying a React SPA (S3 bucket, cloudfront, API). I have everything contained within a stage. The issue I'm having is this requires an S3 deployment action. I want to keep things encapsulated and make it simple to deploy to multiple environments but I cannot find a way of having the S3 bucket within a stage and also using it in a deploy action without the following `dependency cannot cross stage boundaries`.

### Environment

  - **CDK CLI Version:** 1.70
  - **Module Version:** 1.70
  - **Node.js Version:** v15.1.0
  - **OS:** Ubuntu
  - **Language (Version):** TypeScript (3.8.3)

### PR DESCRIPTION
**Pull request by rix0rrr**
To transport values across Stages, users need to construct their
own `{ Fn::ImportValue }` expressions, as we cannot properly do this
for them in order to keep Stages deterministic and isolated.

Add an expression constructor to make this API easier to use.

Resolves #11360.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/core/lib/cfn-output.ts [hunk 1]
```diff
@@ -132,4 +162,6 @@ export class CfnOutput extends CfnElement {
   }
 }
 
-import { CfnCondition } from './cfn-condition';
+import { CfnCondition } from './cfn-condition';import { Fn } from './cfn-fn';
+import { Lazy } from './lazy';
```
Comment (eladb) [inline packages/@aws-cdk/core/lib/cfn-output.ts:166]:
```suggestion
import { CfnCondition } from './cfn-condition';
import { Fn } from './cfn-fn';
import { Lazy } from './lazy';
```

File: packages/@aws-cdk/core/lib/cfn-output.ts [hunk 2]
```diff
@@ -115,6 +116,35 @@ export class CfnOutput extends CfnElement {
...
+   * You can use this mechanism to share values across Stacks in different
+   * Stages. If you intend to share the value to another Stack inside the same
+   * Stage, the automatic cross-stack referencing mechanism is more convenient.
+   */
+  public get importValue() {
+    // We made _exportName mutable so this will have to be lazy.
+    return Fn.importValue(Lazy.stringValue({
+      produce: () => {
+        if (!this._exportName) {
+          throw new Error(`Add an exportName to the CfnOutput at '${this.node.path}' in order to use 'output.importValue'`);
```
Comment (eladb) [inline packages/@aws-cdk/core/lib/cfn-output.ts:140]: Can't we auto-allocate an export name in this case (similar to the one we allocate for auto-refs)?

File: packages/@aws-cdk/core/lib/cfn-output.ts [hunk 3]
```diff
@@ -115,6 +116,35 @@ export class CfnOutput extends CfnElement {
     this._exportName = exportName;
   }
 
+  /**
+   * Return the `Fn.importValue` expression to import this value into another stack
+   *
+   * The returned value should not be used in the same stack, but in a
```
Comment (eladb) [inline packages/@aws-cdk/core/lib/cfn-output.ts:122]: Can we test that this is referenced in the same stack and fail early?

File: packages/@aws-cdk/core/lib/cfn-output.ts (hunk 2 above)
Comment (rix0rrr) [reply packages/@aws-cdk/core/lib/cfn-output.ts:140]: We could, but I'm not 100% comfortable doing that yet. Doesn't seem too onerous to start by expecting people to provide a name.

Comment (eladb) [reply packages/@aws-cdk/core/lib/cfn-output.ts:143]: ok

**Review by eladb — APPROVED** [review]

**Pull request merged**